src/server/gc_worker/gc_worker.rs

```diff
@@ -537,35 +538,34 @@ where
     ) -> Result<()> {
         let start_key = key.clone().append_ts(safe_point.prev());
         let mut cursor = CursorBuilder::new(kv_snapshot, CF_DEFAULT).build()?;
-
         let mut statistics = CfStatistics::default();
-
         cursor.seek(&start_key, &mut statistics)?;
 
-        let mut is_latest_version = true;
+        let mut remove_older = false;
+        let mut latest_version_key: Vec<u8> = vec![];
 
         while cursor.valid()? {
             let engine_slice = cursor.key(&mut statistics);
             if !Key::is_user_key_eq(engine_slice, key.clone().as_encoded()) {
                 break;
             }
 
-            // Skip the latest version before the safe point
-            if is_latest_version {
-                is_latest_version = false;
-                cursor.next(&mut statistics);
-                continue;
-            }
-
             if raw_modifies.write_size >= MAX_RAW_WRITE_SIZE {
                 return Ok(());
             }
 
-            let engine_key = Key::from_encoded_slice(engine_slice);
-            let write = Modify::Delete(CF_DEFAULT, engine_key);
+            if remove_older {
+                self.delete_raw_write(engine_slice.to_vec(), raw_modifies);
+            } else {
+                remove_older = true;
 
-            raw_modifies.write_size += write.size();
-            raw_modifies.modifies.push(write);
+                let value = ApiV2::decode_raw_value(cursor.value(&mut statistics))?;
+                let current_ts = ttl_current_ts();
```

Comment:
You can get the ts outside the loop, or outside the function `raw_gc_key` so that it can be better tested.

Reply:
done